### ISSUE TITLE
order of methods in files

### ISSUE THREAD

**Issue by madscatt** [madscatt/zazmol#31]
Need to alphabetize the methods so that the ordering in code matches ordering in sphinx documentation.

**Comment by madscatt** [comment]
Short answer: don’t reorder your source code to satisfy docs. Make Sphinx do the ordering.

---

We should not alphabetize methods in the source just to match documentation — that’s brittle and will keep breaking as methods are added. The right place to enforce ordering is in the Sphinx layer.

Sphinx (via autodoc) already supports controlling member order. We can set:

```python
autodoc_member_order = 'alphabetical'
```

in `conf.py`, which will alphabetize methods in the generated docs regardless of their order in the source.

If we want more control, we can also:

* Use `'groupwise'` to keep logical groupings (e.g., methods vs attributes) while still being predictable
* Use explicit `:members:` lists in directives for critical classes where we want curated ordering

For edge cases or future flexibility, we could add a lightweight extension hook (e.g., `autodoc-process-docstring` or `autodoc-skip-member`) to enforce custom ordering rules without touching source.

Bottom line: keep code organized for readability and logic, and let Sphinx handle presentation order.
